presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloSplitManager.java

```diff
@@ -0,0 +1,172 @@
+import io.airlift.log.Logger;
+
+import javax.inject.Inject;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static com.facebook.presto.accumulo.Types.checkType;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Presto split manager for Accumulo. Main entry point for logically splitting an Accumulo table
+ * into multiple splits for a parallel data read
+ */
+public class AccumuloSplitManager
+        implements ConnectorSplitManager
+{
+    private static final Logger LOG = Logger.get(AccumuloSplitManager.class);
+
+    private final String connectorId;
+    private final AccumuloClient client;
+
+    /**
+     * Creates a new instance of an {@link AccumuloSplitManager}, injected by the Google.
+     *
+     * @param connectorId Connector ID
+     * @param client Client object for retrieving table metadata and splits
+     */
+    @Inject
+    public AccumuloSplitManager(AccumuloConnectorId connectorId, AccumuloClient client)
+    {
+        this.connectorId = requireNonNull(connectorId, "connectorId is null").toString();
+        this.client = requireNonNull(client, "client is null");
+    }
+
+    /**
+     * Gets the source of splits from the given Presto table layout
+     *
+     * @param session Current client session
+     * @param layout Table layout
+     * @return Split source to pass splits to Presto for scan
+     */
+    @Override
+    public ConnectorSplitSource getSplits(ConnectorSession session,
+            ConnectorTableLayoutHandle layout)
+    {
+        AccumuloTableLayoutHandle layoutHandle =
+                checkType(layout, AccumuloTableLayoutHandle.class, "layout");
+        AccumuloTableHandle tableHandle = layoutHandle.getTable();
+
+        String schemaName = tableHandle.getSchema();
+        String tableName = tableHandle.getTable();
+        String rowIdName = tableHandle.getRowId();
+
+        // If the column filter optimization is enabled, then get the list of column constraints to
+        // pack into a split
+        // These constraints are later converted to an Accumulo iterator and run on the tablet
+        // server
+        List<AccumuloColumnConstraint> iteratorConstraints;
+        if (AccumuloSessionProperties.isOptimizeColumnFiltersEnabled(session)) {
+            iteratorConstraints = getColumnConstraints(rowIdName, layoutHandle.getConstraint());
+        }
+        else {
+            iteratorConstraints = ImmutableList.of();
+        }
+
+        // Get the row domain column range
+        Domain rDom = getRangeDomain(rowIdName, layoutHandle.getConstraint());
+
+        // Call out to our client to retrieve all tablet split metadata using the row ID domain
+        // and the secondary index, if enabled and proper to do so
+        List<TabletSplitMetadata> tSplits = client.getTabletSplits(session, schemaName, tableName,
+                rDom, getColumnConstraints(rowIdName, layoutHandle.getConstraint()),
```

Comment:
Can you extract `getColumnConstraints`? You seem to call it twice (when `AccumuloSessionProperties.isOptimizeColumnFiltersEnabled(session)` is `true`).